### DISCUSSION TITLE
[SQL Lab --> Explore] Better Error message when virtual dataset name exists

### DISCUSSION THREAD

**Discussion by zuzana-vej** [apache/superset#18459] [Ideas]
SQL Lab now displays a popup window to let user name their virtual datasource as a part of SQL Lab --> Explore flow. YAY! :) 
When use adds a virtual dataset name which already exists, there is a small error at the bottom right of the page which just says "An Error occurred saving dataset". This is a bit confusing / doesn't help user to address the issue.
<img width="288" alt="Screen Shot 2021-03-10 at 1 37 58 PM" src="https://user-images.githubusercontent.com/61221714/110701429-026a3080-81a6-11eb-95f8-96c5a423cbcb.png">

When user clicks on "Explore" again, same popup comes up, and user is able to paste the already existing name and able to get to Explore, though the name of the virtual dataset is untitled. 

Proposed user experience:
1. Ideally, the error message would state the reason "This name already exists, please select a unique name." OR give user option to override that virtual dataset.
2. If user clicks 2nd time, similar experience should probably happen as 1st time (as long as use knows how to fix the name) or they should have option to proceed with "untitled query" or other random name.

**Comment by junlincc** [comment]
@yousoph ^^.
